src/nmqtt.nim

```diff
@@ -337,11 +344,11 @@ proc onPublish(ctx: MqttCtx, pkt: Pkt) {.async.} =
 
 proc onPubAck(ctx: MqttCtx, pkt: Pkt) {.async.} =
   let (msgId, _) = pkt.getu16(0)
-  assert msgId in ctx.workQueue
-  assert ctx.workQueue[msgId].wk == PubWork
-  assert ctx.workQueue[msgId].state == WorkSent
-  assert ctx.workQueue[msgId].qos == 1
-  ctx.workQueue.del msgId
+  if msgId in ctx.workQueue:
```

Comment:
Instead of `assert`. This is due to the `msgId` could have been deleted in the `work()`, if the connection has been reestablished.